src/metabase/upload.clj

```diff
@@ -68,6 +70,29 @@
   (let [column-limit (some-> driver driver/column-name-length-limit)]
     (min-safe column-limit (max-bytes :model/Field :name))))
 
+(defn truncate-utf8-bytes
```

Comment:
This is complex and a bit slow, but we should almost never need to execute the no-op branch. Perhaps there's something built in that I just couldn't find.